public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/MetricNameField.tsx

```diff
@@ -1,56 +1,24 @@
-import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import React, { useCallback, useMemo } from 'react';
 import { Select } from '@grafana/ui';
 import { SelectableValue } from '@grafana/data';
 
 import { Field } from '../Field';
-import { findOption, toOption } from '../../utils/common';
 import { AzureQueryEditorFieldProps, AzureMonitorOption } from '../../types';
+import { setMetricName } from './setQueryValue';
 
-const ERROR_SOURCE = 'metrics-metricname';
-const MetricName: React.FC<AzureQueryEditorFieldProps> = ({
-  query,
-  datasource,
-  subscriptionId,
-  variableOptionGroup,
-  onQueryChange,
-  setError,
-}) => {
-  const [metricNames, setMetricNames] = useState<AzureMonitorOption[]>([]);
-  const [isLoading, setIsLoading] = useState(false);
-
-  useEffect(() => {
-    const { resourceGroup, metricDefinition, resourceName, metricNamespace } = query.azureMonitor ?? {};
-    if (!(subscriptionId && resourceGroup && metricDefinition && resourceName && metricNamespace)) {
-      metricNames.length > 0 && setMetricNames([]);
-      return;
-    }
-    setIsLoading(true);
-
-    datasource
-      .getMetricNames(subscriptionId, resourceGroup, metricDefinition, resourceName, metricNamespace)
-      .then((results) => {
-        setMetricNames(results.map(toOption));
-        setIsLoading(false);
-      })
-      .catch((err) => {
-        setError(ERROR_SOURCE, err);
-        setIsLoading(false);
-      });
-  }, [datasource, metricNames.length, query.azureMonitor, setError, subscriptionId]);
+interface MetricNameProps extends AzureQueryEditorFieldProps {
+  metricNames: AzureMonitorOption[];
+}
 
+const MetricNameField: React.FC<MetricNameProps> = ({ metricNames, query, variableOptionGroup, onQueryChange }) => {
```

Comment:
This is a stupid question probably but is there a meaningful difference between this and 

```
const MetricNameField = ({ metricNames, query, variableOptionGroup, onQueryChange }: MetricNameProps) => {
```

Reply:
meaningful? No, not really. There are some differences which you may or may not think are meaningful https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/

And I actually think its house style to _not_ use React.FC...

Reply:
I'm going to leave this as is - all of the azure monitor plugin uses React.FC, and there's no major drawback for using it.